shared/lsif/references.ts

```diff
@@ -0,0 +1,154 @@
+import * as sourcegraph from 'sourcegraph'
+import gql from 'tagged-template-noop'
+import { queryGraphQL as sgQueryGraphQL, QueryGraphQLFn } from '../util/graphql'
+import { concat } from '../util/ix'
+import { nodeToLocation, LocationConnectionNode } from './locations'
+import { queryLSIF, GenericLSIFResponse } from './api'
+
+/**
+ * The maximum number of chained GraphQL requests to make for a single
+ * requests query. The page count for a result set should generally be
+ * relatively low unless it's a VERY popular library and LSIF data is
+ * ubiquitous (which is our goal).
+ */
+export const MAX_REFERENCE_PAGE_REQUESTS = 10
+
+export interface ReferencesResponse {
+    references: {
+        nodes: LocationConnectionNode[]
+        pageInfo: { endCursor?: string }
+    }
+}
+
+const referencesQuery = gql`
+    query References(
+        $repository: String!
+        $commit: String!
+        $path: String!
+        $line: Int!
+        $character: Int!
+        $after: String
+    ) {
+        repository(name: $repository) {
+            commit(rev: $commit) {
+                blob(path: $path) {
+                    lsif {
+                        references(
+                            line: $line
+                            character: $character
+                            after: $after
+                        ) {
+                            nodes {
+                                resource {
+                                    path
+                                    repository {
+                                        name
+                                    }
+                                    commit {
+                                        oid
+                                    }
+                                }
+                                range {
+                                    start {
+                                        line
+                                        character
+                                    }
+                                    end {
+                                        line
+                                        character
+                                    }
+                                }
+                            }
+                            pageInfo {
+                                endCursor
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+`
+
+/** Retrieve references for the current hover position. */
+// eslint-disable-next-line @typescript-eslint/require-await
```

Comment:
Btw, this issue is fixed if you upgrade to the latest `@sourcegraph/eslint-config` version :)

https://github.com/typescript-eslint/typescript-eslint/issues/1691

Reply:
Will deal with a bunch of renovate stuff next.